### DISCUSSION TITLE
Duplicate env entries in pod spec

### DISCUSSION THREAD

**Discussion by bkrugerp99** [actions/actions-runner-controller#506] [Questions] [answered]
Hi,

Seeing an error in our api logs: 
```
E0312 20:30:58.181762       1 fieldmanager.go:159] [SHOULD NOT HAPPEN] failed to update managedFields for /, Kind=: failed to convert new object (/v1, Kind=Pod) to smd typed: .spec.containers[name="runner"].env: duplicate entries for key [name="RUNNER_WORKDIR"]
```

Looking at our runner deployment, we just have it defined once.  Pods are getting this for env vars: 
```
      - name: RUNNER_WORKDIR
        value: /runner/_work
...
      - name: RUNNER_WORKDIR
        value: /tmp/github-runner
```

Looks like somewhere around here code-wise is my guess: 
https://github.com/summerwind/actions-runner-controller/blob/2273b198a162bdbb602ffbc158748ad063ecc145/controllers/runner_controller.go#L420-L473

I am however not a go programmer.

**Comment by mumoshu** [comment, marked as answer]
@bkrugerp99 Hey! You can use `RunnerDeployment.Spec.Template.Spec.WorkDir` for changing `RUNNER_WORKDIR`. Would it resolve your original issue?

**Comment by mumoshu** [comment]
Please let me move this to Discussiosn as the question is already answered and the feature supports the use case already exists.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
